### PR TITLE
Support for ordering address components

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Added AddressConfigDAO to specify ordering for address components (Street Name, Street Number, Suite)
- Added  correct component order for Brazil, Canada and US

### CODE REVIEW

File: src/foam/nanos/auth/AddressConfig.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AddressConfig.js:3]: 2021

File: src/foam/nanos/auth/AddressConfig.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'AddressConfig',
+
+  ids: ['countryId'],
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AddressConfig.js:11]: Not needed. Will be the first property by default.

File: src/foam/nanos/auth/AddressConfig.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'AddressConfig',
+
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AddressConfig.js:10]: Please add documentation:

File: src/foam/nanos/auth/AddressConfig.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  properties: [
+    {
+      class: 'Reference',
+      targetDAOKey: 'countryDAO',
+      name: 'countryId',
+      of: 'foam.nanos.auth.Country'
+    },
+    {
+      class: 'String',
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AddressConfig.js:21]: Should these be Strings if you're storing numbers for ordering in them?

File: src/foam/nanos/auth/AddressConfig.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  package: 'foam.nanos.auth',
+  name: 'AddressConfig',
+
+  ids: ['countryId'],
+
+  properties: [
+    {
+      class: 'Reference',
+      targetDAOKey: 'countryDAO',
+      name: 'countryId',
```
Comment (jlhughes) [inline src/foam/nanos/auth/AddressConfig.js:17]: rename to country.  our convention for references/relationships. 
as we create  AddressConfig.findCountry(x) which returns the country object.

**Pull request merged**